qcfractal/tests/test_mongoengine_socket.py

```diff
@@ -0,0 +1,74 @@
+"""
+    Unit tests of the mongoengine interface to MongoDB
+
+    Does not use portal or fractal server interfaces.
+
+"""
+
+import pytest
+import qcfractal.interface as portal
+
+from qcfractal.storage_sockets.mongoengine_socket import MongoengineSocket
+from qcfractal.storage_sockets.models import Molecule, Result, Procedure
+import mongoengine as db
+
+
+@pytest.fixture(scope='module')
+def mongoengine_socket():
+    # Drop DB if it exists
+    db_client = db.connect('test_qc_mongoengine')
+    db_client.drop_database('test_qc_mongoengine')
+
+    # connect to the DB using MongoengineSocket
+    mongoengine_socket = MongoengineSocket("mongodb://localhost", 'test_qc_mongoengine')
+
+    yield mongoengine_socket
+
+    # clean up, close connection, and delete Database
+    mongoengine_socket.mongoengine_client.close()
+    # mongoengine_socket.mongoengine_client.drop_database('test_qc_mongoengine')
+
+
+def test_molecule(mongoengine_socket):
+    """
+        Test the use of the ME class Molecule
+
+        Note:
+            creation of a Molecule using ME is not implemented yet
+            Should create a Molecule using: mongoengine_socket.add_molecules
+    """
+
+    # don't use len(Molecule.objects), slow
+    num_mol_in_db = Molecule.objects().count()
```

Comment:
It appears that MongoEngine using a global Mongo client here?

Reply:
Yes.